api/tests/integration/infrastructure/repositories/authentication-method-repository_test.js

```diff
@@ -94,4 +94,110 @@ describe('Integration | Repository | AuthenticationMethod', () => {
+
+      // then
+      expect(authenticationMethodsByUserIdAndIdentityProvider).to.be.instanceof(AuthenticationMethod);
+      expect(authenticationMethodsByUserIdAndIdentityProvider).to.deep.equal(garAuthenticationMethodInDB);
+    });
+
+    it('should return null if there is no AuthenticationMethods for the given user and identity provider', async () => {
+      // given
+      const userId = databaseBuilder.factory.buildUser().id;
+      await databaseBuilder.commit();
+      const identityProvider = AuthenticationMethod.identityProviders.GAR;
+
+      // when
+      const authenticationMethodsByUserIdAndIdentityProvider = await authenticationMethodRepository.findOneByUserIdAndIdentityProvider({ userId, identityProvider });
+
+      // then
+      expect(authenticationMethodsByUserIdAndIdentityProvider).to.be.null;
+    });
+  });
+
+  describe('#findOneByExternalIdentifierAndIdentityProvider', () => {
+
+    it('should return the AuthenticationMethod for a given external identifier and identity provider', async () => {
+      // given
+      const identityProvider = AuthenticationMethod.identityProviders.GAR;
+      const externalIdentifier = 'samlId';
+      const authenticationMethodInDB = databaseBuilder.factory.buildAuthenticationMethod({ externalIdentifier, identityProvider });
+      databaseBuilder.factory.buildAuthenticationMethod({ externalIdentifier: 'another_sub', identityProvider });
+      await databaseBuilder.commit();
+
+      // when
+      const authenticationMethodsByTypeAndValue = await authenticationMethodRepository.findOneByExternalIdentifierAndIdentityProvider({ externalIdentifier, identityProvider });
+
+      // then
+      expect(authenticationMethodsByTypeAndValue).to.be.instanceof(AuthenticationMethod);
+      expect(authenticationMethodsByTypeAndValue).to.deep.equal(authenticationMethodInDB);
+    });
+
+    it('should return null if there is no AuthenticationMethods for the given external identifier and identity provider', async () => {
+      // given
+      const identityProvider = AuthenticationMethod.identityProviders.GAR;
+      const externalIdentifier = 'samlId';
+
+      // when
+      const authenticationMethodsByTypeAndValue = await authenticationMethodRepository.findOneByExternalIdentifierAndIdentityProvider({ externalIdentifier, identityProvider });
+
+      // then
+      expect(authenticationMethodsByTypeAndValue).to.be.null;
+    });
+  });
+
+  describe('#updateExternalIdentifierByUserIdAndIdentityProvider', () => {
+
+    context('When authentication method exists', async () => {
+
+      let authenticationMethod;
+
+      beforeEach(() => {
+        const userId = databaseBuilder.factory.buildUser().id;
+        authenticationMethod = databaseBuilder.factory.buildAuthenticationMethod({ identityProvider: AuthenticationMethod.identityProviders.GAR, userId });
```

Comment:
Puisqu'on teste explicitement la mise à jour de l'externalIdentifier, on pourrait le renseigner à l'appel du builder: 

```suggestion
        authenticationMethod = databaseBuilder.factory.buildAuthenticationMethod({ 
        identityProvider: AuthenticationMethod.identityProviders.GAR, 
        externalIdentifier: `to_be_updated`,
        userId
         });
```

Reply:
Ok